File: tool/table/client/cardbuff.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let cardbuff_map = null;
export function cardbuff_map_init(config_obj:Object):void{
	cardbuff_map = config_obj["cardbuff_map"];
}


export class Cardbuff extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = cardbuff_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Cardbuff(key){
		if(Cardbuff.m_static_map.hasOwnProperty(key) == false){
			Cardbuff.m_static_map[key] = Cardbuff.create_Cardbuff(key);
		}
		return Cardbuff.m_static_map[key];	
	}
	public static create_Cardbuff(key){
		if(cardbuff_map.hasOwnProperty(key)){
			return new Cardbuff(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return cardbuff_map;
	}
}
}

File: tool/table/server/cardbuff.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


cardbuff_map = {};
cardbuff_map[1001] = {"id":1001,"btype":0,"effect":"actor['hpmax']*=1.2","group":1,};
cardbuff_map[1002] = {"id":1002,"btype":0,"effect":"actor['spd']+=20","group":1,};
cardbuff_map[1003] = {"id":1003,"btype":0,"effect":"actor['atk']+=20","group":1,};
cardbuff_map[1004] = {"id":1004,"btype":0,"effect":"actor['def']+=20","group":1,};


class Cardbuff:
	def __init__(self, key):
		config = cardbuff_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Cardbuff(key):
		config = cardbuff_map.get(key);
		if not config:
			return
		return Cardbuff(key)

